### PR TITLE
Jdp210201-05

### PR DESCRIPTION
**Pull request by Ktubbo**
- Dodany UserController, 
- Dodany UserDto,
- Dodany do specs ERD Entity Diagram

### CODE REVIEW

**Review by wisniewskimateusz — CHANGES_REQUESTED** [review]
1. createUser() why PUT instead od POST?
2. createUser() nothing return for tests, why it is empty?

**Comment by Ktubbo** [comment]
>     1. createUser() why PUT instead od POST?
>     2. createUser() nothing return for tests, why it is empty?
1. PUT and POST change its places
2. Return nothing, because it is void method. In postman server will return empty Json.

**Comment by Bauaser-kun** [comment]
generateRandomKey - opis w specyfikacji endpointu mówi o wygenerowaniu klucza po podaniu danych użytkownika, na moje metoda powinna zwracać Long i wymagać parametrów z UserDto. (@RequestBody UserDto)

File: src/main/java/com/kodilla/ecommercee/user/controller/UserController.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    @RequestMapping(method = RequestMethod.POST, value = "createUser")
+    public void createUser(UserDto userDto) { }
+
+    @RequestMapping(method = RequestMethod.PUT, value = "banUser")
+    public UserDto banUser(UserDto userDto) {
+        return new UserDto(1L,"Piotr", 1,59403L);
+    }
+
+    @RequestMapping(method = RequestMethod.PUT, value = "generateRandomKey")
+    public UserDto generateRandomKey(UserDto userDto) {
```
Comment (Bauaser-kun) [inline src/main/java/com/kodilla/ecommercee/user/controller/UserController.java]: generateRandomKey - opis w specyfikacji endpointu mówi o wygenerowaniu klucza po podaniu danych użytkownika, na moje metoda powinna zwracać Long i wymagać parametrów z UserDto. (@ RequestBody UserDto)
And to not get rusty with English once again the same ;P: endpoints specs states about generating new key based on User's data, as for me it means the method should return Long and request params from UserDto.

Comment (Ktubbo) [reply src/main/java/com/kodilla/ecommercee/user/controller/UserController.java]: I have to remind that is only EMPTY cotroller, sending json in postman returns userDto with "new random key" in last field. There is no need to return just long while next you should put it in User field. This method already returns "random key" in User field.

**Comment by Ktubbo** [comment]
> generateRandomKey - opis w specyfikacji endpointu mówi o wygenerowaniu klucza po podaniu danych użytkownika, na moje metoda powinna zwracać Long i wymagać parametrów z UserDto. (@ RequestBody UserDto)
> And to not get rusty with English once again the same ;P: endpoints specs states about generating new key based on User's data, as for me it means the method should return Long and request params from UserDto.

**Comment by Ktubbo** [comment]
- Added @requestbody annotation,
- Changed RequestMapping methods,
- Added info about created user

**Review by lukkra1989 — APPROVED** [review]

**Review by wisniewskimateusz — APPROVED** [review]

**Pull request merged**